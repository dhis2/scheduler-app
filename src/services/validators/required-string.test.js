import requiredString from './required-string'

describe('requiredString', () => {
    it('should return an error when there is no input', () => {
        const actual = requiredString([])

        expect(actual).toBeInstanceOf(Error)
        expect(actual.message).toMatchSnapshot()
    })

    it('should return an error when input is not a string', () => {
        const actual = requiredString([])

        expect(actual).toBeInstanceOf(Error)
        expect(actual.message).toMatchSnapshot()
    })

    it('should return undefined when input is at least a single character', () => {
        const actual = requiredString('a')

        expect(actual).toBeUndefined()
    })
})
