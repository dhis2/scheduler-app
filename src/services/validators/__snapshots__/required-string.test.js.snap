// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`requiredString should return an error when input is not a string 1`] = `"This field is required"`;

exports[`requiredString should return an error when there is no input 1`] = `"This field is required"`;
